packages/grid/x-data-grid-pro/src/components/DataGridProVirtualScroller.tsx

```diff
@@ -387,6 +387,7 @@ const DataGridProVirtualScroller = React.forwardRef<
 
   return (
     <GridVirtualScroller {...getRootProps(other)}>
+      <GridHeaders contentProps={contentProps} ColumnHeadersProps={ColumnHeadersProps} />
```

Comment:
Column resizing regressed a bit because of the DOM structure change:
https://deploy-preview-10059--material-ui-x.netlify.app/x/react-data-grid/column-dimensions/#resizing

https://github.com/mui/mui-x/assets/13808724/a78ca03d-0c58-45d4-9bb2-c2f62b9f8884